### ISSUE TITLE
src/curses/ui_manager.c does not compile when CFLAGS has -Werror=format-security 

### ISSUE THREAD

**Issue by rakotomandimby** [irontec/sngrep#379]
https://github.com/irontec/sngrep/blob/master/src/curses/ui_manager.c does not compile when `CFLAGS` has `-Werror=format-security `

At least Debian and Archlinux default  `CFLAGS` has `-Werror=format-security` and get this when building.

```
curses/ui_manager.c: In function ‘dialog_confirm’:
curses/ui_manager.c:668:5: error: format not a string literal and no format arguments [-Werror=format-security]
  668 |     mvwprintw(dialog_win, 1, (width - strlen(title)) / 2, title);
      |     ^~~~~~~~~
```

Might it be a good idea to fix here (I mean upstream)?

**Comment by Kaian** [comment]
I agree, this should be fixed in the original source code.

I have been trying to reproduce this error without success the past days (I fixed other warnings in latest commits, but not this one). Even with debug enabled (-Wall -Werror) the error does not appear.

<details>
  <summary>Build logs under Gentoo</summary>

```
➜  sngrep git:(master) CFLAGS=-Werror=format-security ./configure --disable-silent-rules       
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking whether ln -s works... yes
checking for egrep... (cached) /bin/grep -E
checking for pthread_create in -lpthread... yes
checking for pcap_open_offline in -lpcap... yes
checking pcap.h usability... yes
checking pcap.h presence... yes
checking for pcap.h... yes
checking ncurses.h usability... yes
checking ncurses.h presence... yes
checking for ncurses.h... yes
checking for initscr in -lncurses... yes
checking for new_panel in -lpanel... yes
checking for new_form in -lform... yes
checking for new_item in -lmenu... yes
checking for library containing keyname... -ltinfo
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes

 ██╗██████╗  ██████╗ ███╗   ██╗████████╗███████╗ ██████╗
 ██║██╔══██╗██╔═══██╗████╗  ██║╚══██╔══╝██╔════╝██╔════╝
 ██║██████╔╝██║   ██║██╔██╗ ██║   ██║   █████╗  ██║     
 ██║██╔══██╗██║   ██║██║╚██╗██║   ██║   ██╔══╝  ██║     
 ██║██║  ██║╚██████╔╝██║ ╚████║   ██║   ███████╗╚██████╗
 ╚═╝╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═══╝   ╚═╝   ╚══════╝ ╚═════╝

configure: 
configure: sngrep configure finished                            	
configure: ====================================================== 	
configure: GnuTLS Support               : no		
configure: OpenSSL Support              : no 			
configure: Unicode Support              : no  		
configure: Perl Expressions Support     : no              
configure: IPv6 Support                 : no               
configure: EEP Support                  : no               
configure: ====================================================== 	
configure: 
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating config/Makefile
config.status: creating doc/Makefile
config.status: creating tests/Makefile
config.status: creating src/config.h
config.status: src/config.h is unchanged
config.status: executing depfiles commands
```


```
➜  sngrep git:(master) make                                                             
Making all in src
make[1]: se entra en el directorio '/home/kaian/projects/sngrep/src'
make  all-am
make[2]: se entra en el directorio '/home/kaian/projects/sngrep/src'
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-capture.o -MD -MP -MF .deps/sngrep-capture.Tpo -c -o sngrep-capture.o `test -f 'capture.c' || echo './'`capture.c
mv -f .deps/sngrep-capture.Tpo .deps/sngrep-capture.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-address.o -MD -MP -MF .deps/sngrep-address.Tpo -c -o sngrep-address.o `test -f 'address.c' || echo './'`address.c
mv -f .deps/sngrep-address.Tpo .deps/sngrep-address.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-packet.o -MD -MP -MF .deps/sngrep-packet.Tpo -c -o sngrep-packet.o `test -f 'packet.c' || echo './'`packet.c
mv -f .deps/sngrep-packet.Tpo .deps/sngrep-packet.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-sip.o -MD -MP -MF .deps/sngrep-sip.Tpo -c -o sngrep-sip.o `test -f 'sip.c' || echo './'`sip.c
mv -f .deps/sngrep-sip.Tpo .deps/sngrep-sip.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-sip_call.o -MD -MP -MF .deps/sngrep-sip_call.Tpo -c -o sngrep-sip_call.o `test -f 'sip_call.c' || echo './'`sip_call.c
mv -f .deps/sngrep-sip_call.Tpo .deps/sngrep-sip_call.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-sip_msg.o -MD -MP -MF .deps/sngrep-sip_msg.Tpo -c -o sngrep-sip_msg.o `test -f 'sip_msg.c' || echo './'`sip_msg.c
mv -f .deps/sngrep-sip_msg.Tpo .deps/sngrep-sip_msg.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-sip_attr.o -MD -MP -MF .deps/sngrep-sip_attr.Tpo -c -o sngrep-sip_attr.o `test -f 'sip_attr.c' || echo './'`sip_attr.c
mv -f .deps/sngrep-sip_attr.Tpo .deps/sngrep-sip_attr.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-main.o -MD -MP -MF .deps/sngrep-main.Tpo -c -o sngrep-main.o `test -f 'main.c' || echo './'`main.c
mv -f .deps/sngrep-main.Tpo .deps/sngrep-main.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-option.o -MD -MP -MF .deps/sngrep-option.Tpo -c -o sngrep-option.o `test -f 'option.c' || echo './'`option.c
mv -f .deps/sngrep-option.Tpo .deps/sngrep-option.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-group.o -MD -MP -MF .deps/sngrep-group.Tpo -c -o sngrep-group.o `test -f 'group.c' || echo './'`group.c
mv -f .deps/sngrep-group.Tpo .deps/sngrep-group.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-filter.o -MD -MP -MF .deps/sngrep-filter.Tpo -c -o sngrep-filter.o `test -f 'filter.c' || echo './'`filter.c
mv -f .deps/sngrep-filter.Tpo .deps/sngrep-filter.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-keybinding.o -MD -MP -MF .deps/sngrep-keybinding.Tpo -c -o sngrep-keybinding.o `test -f 'keybinding.c' || echo './'`keybinding.c
mv -f .deps/sngrep-keybinding.Tpo .deps/sngrep-keybinding.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-media.o -MD -MP -MF .deps/sngrep-media.Tpo -c -o sngrep-media.o `test -f 'media.c' || echo './'`media.c
mv -f .deps/sngrep-media.Tpo .deps/sngrep-media.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-setting.o -MD -MP -MF .deps/sngrep-setting.Tpo -c -o sngrep-setting.o `test -f 'setting.c' || echo './'`setting.c
mv -f .deps/sngrep-setting.Tpo .deps/sngrep-setting.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-rtp.o -MD -MP -MF .deps/sngrep-rtp.Tpo -c -o sngrep-rtp.o `test -f 'rtp.c' || echo './'`rtp.c
mv -f .deps/sngrep-rtp.Tpo .deps/sngrep-rtp.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-util.o -MD -MP -MF .deps/sngrep-util.Tpo -c -o sngrep-util.o `test -f 'util.c' || echo './'`util.c
mv -f .deps/sngrep-util.Tpo .deps/sngrep-util.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-hash.o -MD -MP -MF .deps/sngrep-hash.Tpo -c -o sngrep-hash.o `test -f 'hash.c' || echo './'`hash.c
mv -f .deps/sngrep-hash.Tpo .deps/sngrep-hash.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT sngrep-vector.o -MD -MP -MF .deps/sngrep-vector.Tpo -c -o sngrep-vector.o `test -f 'vector.c' || echo './'`vector.c
mv -f .deps/sngrep-vector.Tpo .deps/sngrep-vector.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_panel.o -MD -MP -MF curses/.deps/sngrep-ui_panel.Tpo -c -o curses/sngrep-ui_panel.o `test -f 'curses/ui_panel.c' || echo './'`curses/ui_panel.c
mv -f curses/.deps/sngrep-ui_panel.Tpo curses/.deps/sngrep-ui_panel.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-scrollbar.o -MD -MP -MF curses/.deps/sngrep-scrollbar.Tpo -c -o curses/sngrep-scrollbar.o `test -f 'curses/scrollbar.c' || echo './'`curses/scrollbar.c
mv -f curses/.deps/sngrep-scrollbar.Tpo curses/.deps/sngrep-scrollbar.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_manager.o -MD -MP -MF curses/.deps/sngrep-ui_manager.Tpo -c -o curses/sngrep-ui_manager.o `test -f 'curses/ui_manager.c' || echo './'`curses/ui_manager.c
mv -f curses/.deps/sngrep-ui_manager.Tpo curses/.deps/sngrep-ui_manager.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_call_list.o -MD -MP -MF curses/.deps/sngrep-ui_call_list.Tpo -c -o curses/sngrep-ui_call_list.o `test -f 'curses/ui_call_list.c' || echo './'`curses/ui_call_list.c
mv -f curses/.deps/sngrep-ui_call_list.Tpo curses/.deps/sngrep-ui_call_list.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_call_flow.o -MD -MP -MF curses/.deps/sngrep-ui_call_flow.Tpo -c -o curses/sngrep-ui_call_flow.o `test -f 'curses/ui_call_flow.c' || echo './'`curses/ui_call_flow.c
mv -f curses/.deps/sngrep-ui_call_flow.Tpo curses/.deps/sngrep-ui_call_flow.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_call_raw.o -MD -MP -MF curses/.deps/sngrep-ui_call_raw.Tpo -c -o curses/sngrep-ui_call_raw.o `test -f 'curses/ui_call_raw.c' || echo './'`curses/ui_call_raw.c
mv -f curses/.deps/sngrep-ui_call_raw.Tpo curses/.deps/sngrep-ui_call_raw.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_stats.o -MD -MP -MF curses/.deps/sngrep-ui_stats.Tpo -c -o curses/sngrep-ui_stats.o `test -f 'curses/ui_stats.c' || echo './'`curses/ui_stats.c
mv -f curses/.deps/sngrep-ui_stats.Tpo curses/.deps/sngrep-ui_stats.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_filter.o -MD -MP -MF curses/.deps/sngrep-ui_filter.Tpo -c -o curses/sngrep-ui_filter.o `test -f 'curses/ui_filter.c' || echo './'`curses/ui_filter.c
mv -f curses/.deps/sngrep-ui_filter.Tpo curses/.deps/sngrep-ui_filter.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_save.o -MD -MP -MF curses/.deps/sngrep-ui_save.Tpo -c -o curses/sngrep-ui_save.o `test -f 'curses/ui_save.c' || echo './'`curses/ui_save.c
mv -f curses/.deps/sngrep-ui_save.Tpo curses/.deps/sngrep-ui_save.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_msg_diff.o -MD -MP -MF curses/.deps/sngrep-ui_msg_diff.Tpo -c -o curses/sngrep-ui_msg_diff.o `test -f 'curses/ui_msg_diff.c' || echo './'`curses/ui_msg_diff.c
mv -f curses/.deps/sngrep-ui_msg_diff.Tpo curses/.deps/sngrep-ui_msg_diff.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_column_select.o -MD -MP -MF curses/.deps/sngrep-ui_column_select.Tpo -c -o curses/sngrep-ui_column_select.o `test -f 'curses/ui_column_select.c' || echo './'`curses/ui_column_select.c
mv -f curses/.deps/sngrep-ui_column_select.Tpo curses/.deps/sngrep-ui_column_select.Po
gcc -DHAVE_CONFIG_H -I.      -Werror=format-security -MT curses/sngrep-ui_settings.o -MD -MP -MF curses/.deps/sngrep-ui_settings.Tpo -c -o curses/sngrep-ui_settings.o `test -f 'curses/ui_settings.c' || echo './'`curses/ui_settings.c
mv -f curses/.deps/sngrep-ui_settings.Tpo curses/.deps/sngrep-ui_settings.Po
gcc   -Werror=format-security   -o sngrep sngrep-capture.o    sngrep-address.o sngrep-packet.o sngrep-sip.o sngrep-sip_call.o sngrep-sip_msg.o sngrep-sip_attr.o sngrep-main.o sngrep-option.o sngrep-group.o sngrep-filter.o sngrep-keybinding.o sngrep-media.o sngrep-setting.o sngrep-rtp.o sngrep-util.o sngrep-hash.o sngrep-vector.o curses/sngrep-ui_panel.o curses/sngrep-scrollbar.o curses/sngrep-ui_manager.o curses/sngrep-ui_call_list.o curses/sngrep-ui_call_flow.o curses/sngrep-ui_call_raw.o curses/sngrep-ui_stats.o curses/sngrep-ui_filter.o curses/sngrep-ui_save.o curses/sngrep-ui_msg_diff.o curses/sngrep-ui_column_select.o curses/sngrep-ui_settings.o   -ltinfo -lmenu -lform -lpanel -lncurses -lpcap -lpthread 
make[2]: se sale del directorio '/home/kaian/projects/sngrep/src'
make[1]: se sale del directorio '/home/kaian/projects/sngrep/src'
Making all in config
make[1]: se entra en el directorio '/home/kaian/projects/sngrep/config'
make[1]: No se hace nada para 'all'.
make[1]: se sale del directorio '/home/kaian/projects/sngrep/config'
Making all in doc
make[1]: se entra en el directorio '/home/kaian/projects/sngrep/doc'
make[1]: No se hace nada para 'all'.
make[1]: se sale del directorio '/home/kaian/projects/sngrep/doc'
Making all in tests
make[1]: se entra en el directorio '/home/kaian/projects/sngrep/tests'
make[1]: No se hace nada para 'all'.
make[1]: se sale del directorio '/home/kaian/projects/sngrep/tests'
make[1]: se entra en el directorio '/home/kaian/projects/sngrep'
make[1]: No se hace nada para 'all-am'.
make[1]: se sale del directorio '/home/kaian/projects/sngrep'

</details>

**Comment by Kaian** [comment]
This should be fixed in #382

Thanks for your report and pull request!!

Regards
